lib/Transforms/Scalar/UnifyAssumes.cc

```diff
@@ -172,10 +172,17 @@ void UnifyAssumesPass::processAssertInst(CallInst &CI, AllocaInst &flag) {
   llvm::CallSite CS(&CI);
   IRBuilder<> B(bb);
   B.SetInsertPoint(&CI);
+  Value *conseq = CS.getArgument(0);
+  // remove instruction if verifier.assert(true)
+  if (auto *conseq_const = dyn_cast<llvm::Constant>(conseq)) {
+    if (!conseq_const->isZeroValue()) {
+      CI.eraseFromParent();
```

Comment:
are you sure this does not invalidate any iterator?

Reply:
Are we running on multithreaded mode? I think the iterator only affected is here:
https://github.com/seahorn/seahorn/blob/dev10/lib/Transforms/Scalar/UnifyAssumes.cc#L141-L143
Is that ok I created a new vector to store `assert(true)` instruction and delete them at once after invoking `processAssertInst`?

Reply:
Oh, all is good. We are traversive the `asserts` and deleting them not form the list we are traversing.
Just empty the `asserts` vector after the for loop since it might contain deallocated memory. We should not traverse the asserts more than once.

Reply:
Added cleanup code after used for `assert` vector on new commit.